src/bsoncxx/test/view_or_value.cpp

```diff
@@ -0,0 +1,34 @@
+#include "catch.hpp"
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/document/view_or_value.hpp>
+#include <bsoncxx/json.hpp>
+
+using bsoncxx::to_json;
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::finalize;
+using bsoncxx::document::view_or_value;
+
+TEST_CASE("view_or_value", "[bsoncxx::document::view_or_value]") {
+    auto doc = document{} << "a" << 1 << finalize;
+    auto json = to_json(doc.view());
+
+    SECTION("can be constructed with a view") {
+        view_or_value variant(doc.view());
+
+        SECTION("can be used as a view") {
+            REQUIRE(to_json(variant) == json);
+        }
+    }
+
+    SECTION("can be constructed with a value") {
+        view_or_value variant(doc);
```

Comment:
it doesn't make the test wrong, but it shows a usability problem as making that copy somewhat defeats the point of this entire class.